src/platforms/f4discovery/platform.c

```diff
@@ -34,25 +34,41 @@
 #include <libopencm3/stm32/usart.h>
 #include <libopencm3/stm32/syscfg.h>
 #include <libopencm3/usb/usbd.h>
+#include <libopencm3/cm3/systick.h>
+#include <libopencm3/cm3/cortex.h>
 
 jmp_buf fatal_error_jmpbuf;
+extern uint32_t _ebss;
 
 void platform_init(void)
 {
+	volatile uint32_t *magic = (uint32_t *) &_ebss;
 	/* Check the USER button*/
-	rcc_peripheral_enable_clock(&RCC_AHB1ENR, RCC_AHB1ENR_IOPAEN);
-	if(gpio_get(GPIOA, GPIO0)) {
-		platform_request_boot();
+	rcc_periph_clock_enable(RCC_GPIOA);
+	if (gpio_get(GPIOA, GPIO0) ||
+		((magic[0] == BOOTMAGIC0) && (magic[1] == BOOTMAGIC1))) {
```

Comment:
Please fix the indentation here so it lines up with the opening bracket of the if condition.